framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -721,12 +731,39 @@ protected void addInformationFromPreconditions(S info,
 
         S info = in.getRegularStore();
         V rhsValue = in.getValueOfSubNode(rhs);
+
+        Receiver expr = FlowExpressions.internalReprOf(analysis.getTypeFactory(),
+                n.getTarget());
+
+        if (inferSignatures && !expr.containsUnknown()
```

Comment:
Can you add documentation for Receiver.containsUnknown() and containsOfClass, and more importantly justify why there is no special case for type systems that don't use exactly Unknown.class?  What about those that define their own top annotation, such as UnknownKeyFor?  Does it matter whether the Unknown annotation is the type system's default or not?

More generally, I would like to understand what the intention of this code is, so that I can better understand why it is correct.  Thanks!